lib/broccoli/ember-app.js

```diff
@@ -63,21 +63,9 @@ module.exports = EmberApp;
   @constructor
   @param {Object} options Configuration options
 */
-function EmberApp() {
-  var args = [];
-  var options = {};
-
-  for (var i = 0, l = arguments.length; i < l; i++) {
-    args.push(arguments[i]);
-  }
-
-  if (args.length === 1) {
-    options = args[0];
-  } else if (args.length > 1) {
-    args.reverse();
-    options = defaults.apply(null, args);
-  }
 
+function EmberApp() {
+  var options = this._extractOptions(arguments);
```

Comment:
> Doesn't shrinking the big function into smaller ones helps to inline them and optimize it ?

It isn't that straight forward.

Ultimately this function doesn't need to be fast, so it should just be w/e is most clear to read. Ignore my above concerns about unrolling args here.